lib/ldp.js

```diff
@@ -256,25 +256,53 @@ class LDP {
     })
   }
 
-  put (host, resourcePath, stream, callback) {
+  isValidRdf (body, requestUri, contentType) {
+    const resourceGraph = $rdf.graph()
+    try {
+      $rdf.parse(body, resourceGraph, requestUri, contentType)
+    } catch (err) {
+      debug.ldp('VALIDATE -- Error parsing data: ' + err)
+      return false
+    }
+    return true
+  }
+
+  _put (host, resourcePath, callback, resourceHandler) {
     const ldp = this
     const root = !ldp.multiuser ? ldp.root : ldp.root + host + '/'
     const filePath = utils.uriToFilename(resourcePath, root, host)
 
     // PUT requests not supported on containers. Use POST instead
     if (filePath.endsWith('/')) {
-      return callback(error(409,
-        'PUT not supported on containers, use POST instead'))
+      return callback(error(409, 'PUT not supported on containers, use POST instead'))
```

Comment:
Wouldn't be a [405 - Method not allowed](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_errors) be more appropriate then?

Reply:
Definitely 405.